impl/client/src/main/javascript/web/pentaho/ccc/visual/Funnel.js

```diff
@@ -0,0 +1,155 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "pentaho/module!_",
+  "pentaho/visual/impl/View",
+  "common-ui/echarts",
+  "./_util",
+  "pentaho/visual/color/util"
+], function (module, BaseView, echarts, util, visualColorUtils) {
+
+  "use strict";
+
+  return BaseView.extend(module.id, {
+
+    configureLabel: function (option) {
+      var font = util.defaultFont(null, 12);
+
+      var label = {
+        show: true,
+        position: this.model.labelsOption,
+        formatter: '{b}: {d}%',
+        backgroundColor: 'transparent',
+        fontSize: font.substring(0, font.indexOf(' ')),
+        fontFamily: font.substring(font.indexOf(' ') + 1)
+      };
+
+      option.series.forEach(function (row) {
+        row.label = label;
+      })
+    },
+
+    configureLegend: function (option, records) {
+      var categories = [];
+      var font = util.defaultFont(null, 14);
+
+      records.forEach(function (record) {
+          categories.push(record.name);
+        }
+      );
+
+      var legend = {
+        data: categories,
+        type: 'scroll',
+        textStyle: {
+          fontSize: font.substring(0, font.indexOf(' ')),
+          fontFamily: font.substring(font.indexOf(' ') + 1),
+        }
+      };
+      option.legend = legend;
+    },
+
+    configureColors: function () {
+      return this.model.palette.colors.toArray(function (color) {
+        return color.value;
+      });
+    },
+
+    _updateAll: function () {
+      this.base();
+      var myChart = echarts.init(this.__domContainer, null, {});
+
+      var dataTable = this.model.data;
+      if (dataTable.originalCrossTable) {
+        dataTable = dataTable.originalCrossTable.toPlainTable({skipRowsWithAllNullMeasures: true});
+      }
+
+      var rowLength = dataTable.getNumberOfRows();
+      var colLength = dataTable.getNumberOfColumns();
+      var range = dataTable.getColumnRange(colLength - 1);
+      var font = util.defaultFont(null, 12);
+
+      var records = [];
+
+      // Read data from dataTable and push it to records
+      for (var i = 0; i < rowLength; i++) {
+        var tooltipFormatString = '';
+        for (var j = 0; j < colLength; j++) {
+          tooltipFormatString = tooltipFormatString + dataTable.getColumnLabel(j) + " : " + dataTable.getFormattedValue(i, j) + "<br />";
+        }
+        records.push({
+          name: dataTable.getFormattedValue(i, colLength - 2),
+          value: dataTable.getValue(i, colLength - 1),
+          tooltip: {
+            formatter: tooltipFormatString,
+            fontSize: font.substring(0, font.indexOf(' ')),
+            fontFamily: font.substring(font.indexOf(' ') + 1)
+          },
+        });
+      }
+
+      var option = {
```

Comment:
Should be plural, no?

Reply:
Generally this is a single object and hence no. While series is an array.

Reply:
The plural refers to the object being a bag / a set of options. Each property in the object is an option.